precompiles/bank/bank.go

```diff
@@ -169,6 +178,38 @@ func (p Precompile) send(ctx sdk.Context, method *abi.Method, args []interface{}
 	return method.Outputs.Pack(true)
 }
 
+func (p Precompile) sendNative(ctx sdk.Context, method *abi.Method, args []interface{}) ([]byte, error) {
+	pcommon.AssertArgsLength(args, 4)
+	denom := args[2].(string)
+	if denom == "" {
+		return nil, errors.New("invalid denom")
+	}
+	amount := args[3].(*big.Int)
+	if amount.Cmp(big.NewInt(0)) == 0 {
+		// short circuit
+		return method.Outputs.Pack(true)
+	}
+
+	senderSeiAddr, err := p.accAddressFromArg(ctx, args[0])
+	if err != nil {
+		return nil, err
+	}
+
+	receiverAddr, ok := (args[1]).(string)
+	if !ok || receiverAddr == "" {
+		return nil, errors.New("invalid addr")
+	}
+
+	receiverSeiAddr, err := sdk.AccAddressFromBech32(receiverAddr)
+	if err != nil {
+		return nil, err
+	}
+	if err := p.bankKeeper.SendCoins(ctx, senderSeiAddr, receiverSeiAddr, sdk.NewCoins(sdk.NewCoin(denom, sdk.NewIntFromBigInt(amount)))); err != nil {
```

Comment:
The units for this amount must be in 10^6 units?  (usei?)  I wonder whether there's a safety issue here if someone mistakes that within EVM and sends a wei value?

Reply:
That’s a good point. Messaging you to discuss but I believe this also affects the other send precompile